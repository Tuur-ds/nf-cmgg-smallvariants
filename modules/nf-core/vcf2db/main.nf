process VCF2DB {
    tag "$meta.id"
    label 'process_medium'

    // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
    conda "${moduleDir}/environment.yml"
    container "cmgg/vcf2db:2020.02.24"

    input:
    tuple val(meta), path(vcf), path(ped)

    output:
    tuple val(meta), path("*.db") , emit: db
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = "2020.02.24" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    vcf2db.py \\
        $vcf \\
        $ped \\
        ${prefix}.db \\
        $args

    sqlite3 ${prefix}.db 'CREATE INDEX idx_variant_impacts_id ON variant_impacts (variant_id)' && \\
    sqlite3 ${prefix}.db 'ALTER TABLE variants ADD COLUMN tags varchar(255)' && \\
    sqlite3 ${prefix}.db 'ALTER TABLE variants ADD COLUMN tags_user varchar(255)' && \\
    sqlite3 ${prefix}.db 'ALTER TABLE variants ADD COLUMN notes varchar(255)' && \\
    sqlite3 ${prefix}.db 'ALTER TABLE variants ADD COLUMN notes_user varchar(255)'

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vcf2db: $VERSION
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = "2020.02.24" // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    touch ${prefix}.db

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        vcf2db: $VERSION
    END_VERSIONS
    """
}
