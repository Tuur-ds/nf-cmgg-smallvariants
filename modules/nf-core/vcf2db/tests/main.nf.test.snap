{
    "homo_sapiens - vcf, ped - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "caller": "haplotypecaller"
                        },
                        "test.haplotypecaller.db:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3d9eedf7998cb77b614736659eb22b1d"
                ],
                "db": [
                    [
                        {
                            "id": "test",
                            "single_end": false,
                            "caller": "haplotypecaller"
                        },
                        "test.haplotypecaller.db:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3d9eedf7998cb77b614736659eb22b1d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-19T14:40:53.107217359"
    },
    "homo_sapiens - vcf, ped": {
        "content": [
            "test.haplotypecaller.db",
            [
                "versions.yml:md5,3d9eedf7998cb77b614736659eb22b1d"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-19T14:40:45.633504872"
    }
}